pkg/operator/controller/controller.go

```diff
@@ -415,8 +413,13 @@ func (r *reconciler) ensureIngressController(ci *operatorv1.IngressController, d
 			errs = append(errs, fmt.Errorf("failed to integrate metrics with openshift-monitoring for ingresscontroller %s: %v", ci.Name, err))
 		}
 
-		if err := r.syncIngressControllerStatus(deployment, ci); err != nil {
-			errs = append(errs, fmt.Errorf("failed to sync ingresscontroller status: %v", err))
+		operandEvents := &corev1.EventList{}
+		if err := r.cache.List(context.TODO(), operandEvents, client.InNamespace("openshift-ingress")); err != nil {
+			errs = append(errs, fmt.Errorf("failed to list events in namespace %q: %v", "openshift-ingress", err))
```

Comment:
Okay, so this is really subtle and dangerous, but `lbService` can be `nil`. Would love to separately take some action on our prior discussions about:

1. result types vs. nil for dealing with things that are not found
2. coming up with a strategy for dealing with references that doesn't involve hard-coded namespaces or inferring from random resources

Basically I wonder if there's any local fix right here (where's the canonical place to discover the operand namespace in this context?).

Reply:
Oh, snap, good point! Well, does it make sense to look for events if the service does not exist?

Reply:
The lookup isn't dependent on the service, and the lookup is from a cache, so we could choose to provide whatever context we can (consistent with the other possibly-nil inputs).

Reply:
If the service doesn't exist, then `computeLoadBalancerStatus` won't even look at the events, and logically, why would it? Do we anticipate that we might with future changes care about events that are not related to the service?

Reply:
>If the service doesn't exist, then computeLoadBalancerStatus won't even look at the events

`computeLoadBalancerStatus` won't, but this function doesn't know that. This function just knows `computeLoadBalancerStatus` wants events in a namespace. I propose this contract:

```go
// syncIngressControllerStatus takes whatever you give it and updates status. Give it whatever you have and it will do its best.
func (r *reconciler) syncIngressControllerStatus(ic *operatorv1.IngressController, deployment *appsv1.Deployment, service *corev1.Service, operandEvents []corev1.Event) error {
```

😁

Reply:
That seems reasonable... or would it be simpler for `syncIngressControllerStatus` to look up events itself?

Reply:
could perhaps ask the same question regarding all the other arguments... for now unless there's a logic error maybe we can continue the style the discussion on subsequent PRs?

Reply:
> could perhaps ask the same question regarding all the other arguments...

No, `syncIngressControllerStatus` needs to know the ingress controller for which it is updating status, and `ensureIngressController` just created or got the deployment and service, which are sound reasons for `ensureIngressController` to pass those values to `syncIngressControllerStatus`, whereas listing events in `ensureIngressController` instead of in `syncIngressControllerStatus` gratuitously separates the logic of listing events from the logic that determines whether the events need to be listed.

That is not to say that your current approach is unacceptable; the above is only responding to the above-quoted assertion.

That said, if `ensureIngressController` does handling listing events but gets an error, how about passing a nil slice to `syncIngressControllerStatus`?

> for now unless there's a logic error maybe we can continue the style the discussion on subsequent PRs?

That's fine.